source/common/upstream/health_checker_impl.h

```diff
@@ -406,7 +406,7 @@ class RedisHealthCheckerImpl : public HealthCheckerImplBase {
   };
 
   struct HealthCheckRequest {
-    HealthCheckRequest();
+    HealthCheckRequest(const std::string& key);
```

Comment:
Using empty strings to toggle request type is a little wacky. Can you be more explicit and use subclasses (i.e. `PingHealthCheckRequest`)? Or keep the no-arg ctor for PING.

Reply:
i was thinking of using `Optional` actually, thoughts on that vs no-arg ctor?

Reply:
That solves the "empty string is magic" issue but still doesn't indicate what requests you're actually making. No-arg ctor or `HCR(Type, Optional<String>)`?
Where `HealthCheckRequest::Type` is `enum class Type { EXISTS, PING }`?